app/javascript/pages/country/router.js

```diff
@@ -2,27 +2,32 @@ import { connectRoutes } from 'redux-first-router';
 import createHistory from 'history/createBrowserHistory';
 import queryString from 'query-string';
 import { setWidgetSettingsStore } from 'pages/country/widget/widget-actions';
+import { handlePageTrack } from 'utils/analytics';
 
 const history = createHistory();
 
 export const COUNTRY = 'location/COUNTRY';
 export const EMBED = 'location/EMBED';
 
-const syncWidgetSettings = (dispatch, getState) => {
+const routeChangeThunk = (dispatch, getState) => {
+  // sync store with widget settings
   const { query } = getState().location;
   if (query) {
     dispatch(setWidgetSettingsStore(query));
   }
+
+  // track page with GA
+  handlePageTrack(getState().location);
 };
 
 export const routes = {
   [EMBED]: {
     path: '/country/embed/:widget/:country/:region?/:subRegion?',
-    thunk: syncWidgetSettings
+    thunk: routeChangeThunk
```

Comment:
Check the docs of `redux-first-router` there's a config option called `onAfterChange` that sets a thunk for all routes. I think it's better in this case.